indexing-service/src/main/java/io/druid/indexing/common/actions/SurrogateAction.java

```diff
@@ -0,0 +1,89 @@
+/*
+ * Licensed to Metamarkets Group Inc. (Metamarkets) under one
+ * or more contributor license agreements. See the NOTICE file
+ * distributed with this work for additional information
+ * regarding copyright ownership. Metamarkets licenses this file
+ * to you under the Apache License, Version 2.0 (the
+ * "License"); you may not use this file except in compliance
+ * with the License. You may obtain a copy of the License at
+ *
+ * http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing,
+ * software distributed under the License is distributed on an
+ * "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY
+ * KIND, either express or implied. See the License for the
+ * specific language governing permissions and limitations
+ * under the License.
+ */
+package io.druid.indexing.common.actions;
+
+import com.fasterxml.jackson.annotation.JsonCreator;
+import com.fasterxml.jackson.annotation.JsonProperty;
+import com.fasterxml.jackson.core.type.TypeReference;
+import com.google.common.base.Optional;
+import io.druid.indexing.common.task.Task;
+import io.druid.java.util.common.ISE;
+
+/**
+ * Perform the given action using {@link #surrogateId} on behalf of the caller task.
+ */
+public class SurrogateAction<R, T extends TaskAction<R>> implements TaskAction<R>
```

Comment:
nit: I think more descriptive generic type variables than `R` and `T` could make this code a bit easier to read

Reply:
Renamed to `ReturnType` and `ActionType`, respectively.